src/Simplify.elm

```diff
@@ -6163,6 +6182,61 @@ wrapperRepeatChecks wrapper checkInfo =
             Nothing
 
 
+arrayLengthChecks : CheckInfo -> Maybe (Error {})
+arrayLengthChecks checkInfo =
+    firstThatConstructsJust
+        [ \() -> collectionSizeChecks arrayCollection checkInfo
+        , \() -> arrayLengthOnArrayRepeatOrInitializeChecks checkInfo
+        ]
+        ()
+
+
+arrayLengthOnArrayRepeatOrInitializeChecks : CheckInfo -> Maybe (Error {})
+arrayLengthOnArrayRepeatOrInitializeChecks checkInfo =
+    let
+        maybeCall : Maybe ( String, { nodeRange : Range, fnRange : Range, firstArg : Node Expression, argsAfterFirst : List (Node Expression) } )
+        maybeCall =
+            firstThatConstructsJust
+                [ \() ->
+                    AstHelpers.getSpecificFunctionCall ( [ "Array" ], "repeat" ) checkInfo.lookupTable checkInfo.firstArg
+                        |> Maybe.map (Tuple.pair "repeat")
+                , \() ->
+                    AstHelpers.getSpecificFunctionCall ( [ "Array" ], "initialize" ) checkInfo.lookupTable checkInfo.firstArg
+                        |> Maybe.map (Tuple.pair "initialize")
+                ]
+                ()
+    in
+    case maybeCall of
+        Just ( fnName, call ) ->
+            case Evaluate.getInt checkInfo call.firstArg of
```

Comment:
This feels a tiny bit unnecessary because `collectionSizeChecks` will always be called earlier.

Reply:
This is again to prevent having 2 errors. Not sure how to better to do that :thinking:

Reply:
Will remove this check

Reply:
I ended up removing the check but keeping the computation to improve the message and the fix (otherwise we ended up with code like `max 0 -1`)